### ISSUE TITLE
Variables' meaning

### ISSUE THREAD

**Issue by tumbleintoyourheart** [Raschka-research-group/coral-cnn#7]
Dear Sir,

First of all thanks for sharing this great repo

I have some questions. Hopefully, this will not bother you

In `single-image-prediction/afad_coral.py`, can you explain

1. what do `logits` , `probas` and `predicted_levels` mean?
2. why `predicted_label` = `sum(predict_levels)`?

I have had read the paper but still not been able to clarify those

Thank you for your time Sir

_Originally posted by @tumbleintoyourheart in https://github.com/Raschka-research-group/coral-cnn/issues/6#issuecomment-528066010_

**Comment by rasbt** [comment]
> what do logits , probas and predicted_levels mean?

The logits are basically the inverse of the probabilities -- the name comes from logistic regression referring to the "natural logarithm of the odds" (log-odds). In the case of neural networks, people usually call the values before applying the last sigmoid or softmax activation the logits.

The "probas" are the class-membership probabilities then.  The "P"'s here:

![Screen Shot 2019-09-04 at 4 55 19 PM](https://user-images.githubusercontent.com/5618407/64294620-e7758b00-cf34-11e9-88ea-ab5888152058.png)

Predicted levels are the binarized version of that. If the prediction is greater than 0.5, it's 1, and 0 otherwise.

> why predicted_label = sum(predict_levels)

The answer is basically also in the figure above, but to illustrate that with an example:

If you have 5 classes and the true label is 3, you may predict
- P(y > 0): 1
- P(y > 1): 1
- P(y > 2): 1
- P(y > 3): 0

Then, the predicted labels is  1+1+1+0=3.

Hope that helps!

**Comment by tumbleintoyourheart** [comment]
A very concise explanation. I appreciate it. My respect, Sir.

**Comment by tumbleintoyourheart** [comment]
Another one.
* In `training-scripts\afad-coral.py`, `all_pred` is the list of `labels` in `predicted_labels`, and it is written to  `training-results\afad-coral_seed0_imp0\test_predictions.log`. And from what I can see there, they are the actual `predicted age`.
* Meanwhile in `single-image-prediction\afad-coral.py`: `predicted age = predicted label + 15`. I guess `15` is the minimum of "age labels", and `predicted label` here is in the rank ${r_1 = 0, r_2, \dots, r_K}$ set.
What is the difference?

**Comment by rasbt** [comment]
Glad to hear that the explanation was helpful. I guess we can close this then.
